pkg/services/ngalert/store/alertmanager.go

```diff
@@ -110,9 +110,24 @@ func (st DBstore) SaveAlertmanagerConfigurationWithCallback(ctx context.Context,
 // UpdateAlertmanagerConfiguration replaces an alertmanager configuration with optimistic locking. It assumes that an existing revision of the configuration exists in the store, and will return an error otherwise.
 func (st *DBstore) UpdateAlertmanagerConfiguration(ctx context.Context, cmd *models.SaveAlertmanagerConfigurationCmd) error {
 	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
+		newConfigHash := fmt.Sprintf("%x", md5.Sum([]byte(cmd.AlertmanagerConfiguration)))
+		// check for no-op update
+		if newConfigHash == cmd.FetchedConfigurationHash {
+			// double check that the configuration with this hash is in the db
+			ok, err := sess.Table("alert_configuration").
+				Where("org_id = ? AND configuration_hash = ?", cmd.OrgID, cmd.FetchedConfigurationHash).
+				Get(&models.AlertConfiguration{})
```

Comment:
NIT: You can use exists or count instead so we do not fetch data and throw away.

Reply:
Fair point, will update